crates/sui-framework/sources/transfer.move

```diff
@@ -60,13 +31,15 @@ module sui::transfer {
         transfer_internal(obj, recipient, false)
     }
 
+    struct Imm<phantom T: key> has copy, drop { parent: ID, child: ID }
+    struct Mut<phantom T: key> has drop { parent: ID, child: ID }
+    struct Owned<phantom T: key> has drop { parent: ID, child: ID }
```

Comment:
These three types would be the only acceptable form for child objects in `entry` functions. I know its a bit cumbersome, but entry points already likely need to be special for shared objects, because of the presence of the parent object.  So then you will need to do a little dance to borrow/extract the object